kit/Kit.cpp

```diff
@@ -157,6 +159,36 @@ static LokHookFunction2* initFunction = nullptr;
 
 namespace
 {
+    // for later consistency checking.
+    static std::string UserDirPath;
+    static std::string InstDirPath;
+
+    std::string pathFromFileURL(const std::string &uri)
+    {
+        std::string decoded;
+        Poco::URI::decode(uri, decoded);
+        if (decoded.rfind("file://", 0) != 0)
+        {
+            LOG_ERR("Asked to load a very unusual file path: '" << uri << "' -> '" << decoded << "'");
+            return std::string();
+        }
+        return decoded.substr(6);
+    }
+
+    void consistencyCheckFileExists(const std::string &uri)
+    {
+        std::string path = pathFromFileURL(uri);
+        if (path.empty())
+            return;
+        FileUtil::Stat stat(path);
+        if (!stat.good())
+        {
+            LOG_ERR("Fatal system error: created file passed into document doesn't exist: '" << path << "'");
+        }
+        consistencyCheckJail();
+        LOG_TRC("File path '" << path << "' exists of length " << stat.size());
```

Comment:
Perhaps this needs to be under `else`, otherwise in case of `!stat.good()` we'd report that it both does and doesn't exists.